tests/posix/common/src/clock.c

```diff
@@ -109,8 +110,16 @@ void test_posix_realtime(void)
 		/* printk("Delta %d: %lld\n", i, delta); */
 
 		/* Allow for a little drift */
-		zassert_true(error > 90, "Clock inaccurate");
-		zassert_true(error < 110, "Clock inaccurate");
+		/* NOTE: this test is miscalibrated.  It's looking for
+		 * the clock to advance between 90 and 110ms
+		 * (inclusive), but the sleep above is for 90ms.  A
+		 * high tick rate system will nail that delay exactly
```

Comment:
> A
		 * high tick rate system will

Your concern is a high tick rate system, and the concern of whoever wrote that test was a flaming qemu, where you sleep for 1ms, and get 10-20ms passed (though last time I checked, real hw wasn't much better than that). Your misunderstanding of motives behind that code as it is may come from that difference.